### PR TITLE
JAMES-2847 Use BlobStoreDeletedMessageVault in memory product

### PR DESCRIPTION
**Pull request by trantienduchn**
And Drop the MemoryDeletedMessageVault

### CODE REVIEW

File: server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java [hunk 1]
```diff
@@ -826,7 +842,7 @@ void restoreShouldNotAppendMessageToMailboxWhenSubjectDoesntEqualsIgnoreCase() t
 
                 @Test
                 void restoreShouldAppendMessageToMailboxWhenMatchingDeletionDateBeforeOrEquals() throws Exception {
-                    vault.append(DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
+                    Mono.from(Mono.from(vault.append(DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)))).block();
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java]: Why a nested Mono???

File: server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java [hunk 2]
```diff
@@ -858,7 +874,7 @@ void restoreShouldAppendMessageToMailboxWhenMatchingDeletionDateBeforeOrEquals()
 
                 @Test
                 void restoreShouldNotAppendMessageToMailboxWhenNotMatchingDeletionDateBeforeOrEquals() throws Exception {
-                    vault.append(DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
+                    Mono.from(Mono.from(vault.append(DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)))).block();
```
Comment (Arsnael) [inline server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java]: Idem

File: server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java (hunk 1 above)
Comment (trantienduchn) [reply server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java]: probably my find & replace fault

File: server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/java/org/apache/james/jmap/memory/FileMailRepositoryDeletedMessagesVaultTest.java [hunk 3]
```diff
@@ -1,64 +0,0 @@
...
-import org.apache.james.jmap.methods.integration.DeletedMessagesVaultTest;
-import org.apache.james.mailrepository.api.MailRepositoryUrl;
-import org.apache.james.modules.vault.TestDeleteMessageVaultPreDeletionHookModule;
-import org.apache.james.vault.MailRepositoryDeletedMessageVault;
-import org.apache.james.webadmin.WebAdminConfiguration;
-import org.junit.Ignore;
-import org.junit.Rule;
-import org.junit.Test;
-
-public class FileMailRepositoryDeletedMessagesVaultTest extends DeletedMessagesVaultTest {
```
Comment (chibenwa) [inline server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/java/org/apache/james/jmap/memory/FileMailRepositoryDeletedMessagesVaultTest.java]: Why did you delete this file?

File: server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java [hunk 4]
```diff
@@ -370,7 +378,7 @@ void userVaultAPIShouldReturnNotFoundWhenNoUserPathParameter(String action) {
         @ParameterizedTest
         @ValueSource(strings = {"restore", "export"})
         void userVaultAPIShouldReturnBadRequestWhenPassingUnsupportedField(String action) throws Exception {
-            vault.append(DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)).block();
+            Mono.from(Mono.from(vault.append(DELETED_MESSAGE, new ByteArrayInputStream(CONTENT)))).block();
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java]: I don't understand this double Mono.from : is this reeeeeaaally needed?

**Comment by chibenwa** [comment]
Please resolve conflict

File: server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/java/org/apache/james/jmap/memory/FileMailRepositoryDeletedMessagesVaultTest.java (hunk 3 above)
Comment (trantienduchn) [reply server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/java/org/apache/james/jmap/memory/FileMailRepositoryDeletedMessagesVaultTest.java]: this file uses MailRepositoryDeletedMessageVault which relies on the FileMailRepository, and it's inside memory product. Since we switch to use BlobStoreDeletedMessageVault on memory for memory product. This test doesn't add value as I think.

Comment (chibenwa) [reply server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/java/org/apache/james/jmap/memory/FileMailRepositoryDeletedMessagesVaultTest.java]: We still have `MailRepositoryDeletedMessageVault` thus the file composition still needs to be tested. Let's not remove things too fast.

Comment (trantienduchn) [reply server/protocols/jmap-integration-testing/memory-jmap-integration-testing/src/test/java/org/apache/james/jmap/memory/FileMailRepositoryDeletedMessagesVaultTest.java]: then I have to revert the MemoryMailRepositoryDeletedMessagesVaultTest also, I had deleted it

**Comment by trantienduchn** [comment]
test this please

**Comment by trantienduchn** [comment]
```
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/utils/DeleteByQueryExecutorTest.java:[40,37] cannot find symbol
[690c1e52122573b035105cc51915cb5aec340b77]   symbol:   class MemoryDeletedMessagesVault
[690c1e52122573b035105cc51915cb5aec340b77]   location: package org.apache.james.vault.memory
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/utils/DeleteByQueryExecutorTest.java:[50,13] cannot find symbol
[690c1e52122573b035105cc51915cb5aec340b77]   symbol:   class MemoryDeletedMessagesVault
[690c1e52122573b035105cc51915cb5aec340b77]   location: class org.apache.james.vault.utils.DeleteByQueryExecutorTest
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultHookTest.java:[50,37] cannot find symbol
[690c1e52122573b035105cc51915cb5aec340b77]   symbol:   class MemoryDeletedMessagesVault
[690c1e52122573b035105cc51915cb5aec340b77]   location: package org.apache.james.vault.memory
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultHookTest.java:[73,13] cannot find symbol
[690c1e52122573b035105cc51915cb5aec340b77]   symbol:   class MemoryDeletedMessagesVault
[690c1e52122573b035105cc51915cb5aec340b77]   location: class org.apache.james.vault.DeletedMessageVaultHookTest
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/utils/DeleteByQueryExecutorTest.java:[60,33] cannot find symbol
[690c1e52122573b035105cc51915cb5aec340b77]   symbol:   class MemoryDeletedMessagesVault
[690c1e52122573b035105cc51915cb5aec340b77]   location: class org.apache.james.vault.utils.DeleteByQueryExecutorTest
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultHookTest.java:[103,28] cannot find symbol
[690c1e52122573b035105cc51915cb5aec340b77]   symbol:   class MemoryDeletedMessagesVault
[690c1e52122573b035105cc51915cb5aec340b77]   location: class org.apache.james.vault.DeletedMessageVaultHookTest
```

**Comment by trantienduchn** [comment]
Messy relationship between the legacy `DeleteByQueryExecutor` with `DeletedMessageVault`:
 - DeleteByQueryExecutor is requred in DeletedMessageVault for legacy impl (MailRepositoryDeletedMessageVault)
 - DeleteByQueryExecutorTest is using MemoryDeletedMessagesVault which is dropped. the only available alternative is CassandraMailRepositoryDeletedMessageVault, but we cannot import impl module into an API module (that test is in API module)

=> Dead end. This task is easy and could be done in one commit, I will do it after cassandra buckets PR is merged

**Comment by chibenwa** [comment]
```
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/utils/DeleteByQueryExecutorTest.java:[40,37] cannot find symbol
[690c1e52122573b035105cc51915cb5aec340b77]   symbol:   class MemoryDeletedMessagesVault
[690c1e52122573b035105cc51915cb5aec340b77]   location: package org.apache.james.vault.memory
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/utils/DeleteByQueryExecutorTest.java:[50,13] cannot find symbol
[690c1e52122573b035105cc51915cb5aec340b77]   symbol:   class MemoryDeletedMessagesVault
[690c1e52122573b035105cc51915cb5aec340b77]   location: class org.apache.james.vault.utils.DeleteByQueryExecutorTest
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultHookTest.java:[50,37] cannot find symbol
[690c1e52122573b035105cc51915cb5aec340b77]   symbol:   class MemoryDeletedMessagesVault
[690c1e52122573b035105cc51915cb5aec340b77]   location: package org.apache.james.vault.memory
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultHookTest.java:[73,13] cannot find symbol
[690c1e52122573b035105cc51915cb5aec340b77]   symbol:   class MemoryDeletedMessagesVault
[690c1e52122573b035105cc51915cb5aec340b77]   location: class org.apache.james.vault.DeletedMessageVaultHookTest
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/utils/DeleteByQueryExecutorTest.java:[60,33] cannot find symbol
[690c1e52122573b035105cc51915cb5aec340b77]   symbol:   class MemoryDeletedMessagesVault
[690c1e52122573b035105cc51915cb5aec340b77]   location: class org.apache.james.vault.utils.DeleteByQueryExecutorTest
[690c1e52122573b035105cc51915cb5aec340b77] [ERROR] /james-project/mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultHookTest.java:[103,28] cannot find symbol
```
